engine/userguide/storagedriver/aufs-driver.md

```diff
@@ -1,251 +1,236 @@
+  image layer where it exists to the writable container layer. The container
+  then writes the changes to the new copy of the file in the container layer.
 
-The base layer in an image has no image layers below it, so its file is empty.
+  However, AUFS works at the file level rather than the block level. This
+  means that all copy_up operations copy the entire file, even if the file is
+  very large and only a small part of it is being modified. This can have a
+  noticeable impact on container write performance. AUFS, which can suffer
+  noticeable latencies when searching for files in images with many layers.
+  However, it is worth noting that the copy_up operation only occurs the first
+  time a given file is written to. Subsequent writes to the same file operate
+  against the copy of the file already copied up to the container.
 
-### Containers
+- **Deleting files and directories**:
 
-Running containers are mounted below `/var/lib/docker/aufs/mnt/<container-id>`.
- This is where the AUFS union mount point that exposes the container and all
-underlying image layers as a single unified view exists. If a container is not
-running, it still has a directory here but it is empty. This is because AUFS
-only mounts a container when it is running. With Docker 1.10 and higher,
- container IDs no longer correspond to directory names under
-`/var/lib/docker/aufs/mnt/<container-id>`.
+  - When a _file_ is deleted within a container, a *whiteout* file is created
+    in the container layer. The version of the file in the image layer is not
+    deleted (because the image layers are read-only). However, the whiteout
+    file prevents it from being available to the container.
 
-Container metadata and various config files that are placed into the running
-container are stored in `/var/lib/docker/containers/<container-id>`. Files in
-this directory exist for all containers on the system, including ones that are
-stopped. However, when a container is running the container's log files are
-also in this directory.
+  - When a _directory_ is deleted within a container, an _opaque file_ is
+    created in the container layer. This works in the same way as a
+    whiteout file and effectively prevents the directory from being accessed,
+    even though it still exists in the image layer.
 
-A container's thin writable layer is stored in a directory under
-`/var/lib/docker/aufs/diff/`. With Docker 1.10 and higher, container IDs no
-longer correspond to directory names. However, the containers thin writable
-layer still exists here and is stacked by AUFS as the top writable layer
-and is where all changes to the container are stored. The directory exists even
- if the container is stopped. This means that restarting a container will not
-lose changes made to it. Once a container is deleted, its thin writable layer
-in this directory is deleted.
+- **Renaming directories**: Calling `rename(2)` for a directory is not fully
```

Comment:
wat. @mstanleyjones so if one is using aufs, the app will behave differently than with other graph drivers?